staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -740,6 +741,9 @@ func waitForContainer(ctx context.Context, podClient corev1client.PodsGetter, ns
 			if s.State.Running != nil || s.State.Terminated != nil {
 				return true, nil
 			}
+			if s.State.Waiting != nil && s.State.Waiting.Message != "" {
+				fmt.Printf("Container %s: %s\n", containerName, s.State.Waiting.Message)
```

Comment:
For consistency I think this should be:
```suggestion
				fmt.Fprintf(o.Out, "Container %s: %s\n", containerName, s.State.Waiting.Message)
```
but for that you need to pass additional argument to `waitForContainer`.

Reply:
Are these printed to stderr or stdout?

What I'd prefer: stderr
What I'd like as an extra: detect if the console supports color, and if so color these messages so they stand apart from the container's own stderr.

Reply:
It's stdout currently, but I agree that stderr makes more sense. I'll plumb the streams through.

Re: colors, I think that's a cool idea but not one I'd like to take on right now unless there's a library in kubectl that already supports it. I'd be happy to make it clearer textually that the messages is being printed by kubectl, though.

Reply:
(and thanks for these suggestions @sftim & @VirrageS)

Reply:
fixed, please take another look.

Reply:
LGTM

Reply:
Warnings will be overhauled in https://github.com/kubernetes/kubernetes/pull/108210